<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>983</width>
    <height>438</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>983</width>
    <height>430</height>
   </size>
  </property>
  <property name="palette">
   <palette>
    <active>
     <colorrole role="Highlight">
      <brush brushstyle="SolidPattern">
       <color alpha="255">
        <red>252</red>
        <green>175</green>
        <blue>62</blue>
       </color>
      </brush>
     </colorrole>
    </active>
    <inactive>
     <colorrole role="Highlight">
      <brush brushstyle="SolidPattern">
       <color alpha="255">
        <red>252</red>
        <green>175</green>
        <blue>62</blue>
       </color>
      </brush>
     </colorrole>
    </inactive>
    <disabled>
     <colorrole role="Highlight">
      <brush brushstyle="SolidPattern">
       <color alpha="255">
        <red>145</red>
        <green>145</green>
        <blue>145</blue>
       </color>
      </brush>
     </colorrole>
    </disabled>
   </palette>
  </property>
  <property name="windowTitle">
   <string>ExcelFraction</string>
  </property>
  <property name="windowIcon">
   <iconset resource="res.qrc">
    <normaloff>:/image/analys.png</normaloff>:/image/analys.png</iconset>
  </property>
  <property name="autoFillBackground">
   <bool>false</bool>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>641</width>
      <height>71</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout_3">
     <property name="sizeConstraint">
      <enum>QLayout::SetMaximumSize</enum>
     </property>
     <item>
      <widget class="QGroupBox" name="groupBox">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Ignored">
         <horstretch>0</horstretch>
         <verstretch>1</verstretch>
        </sizepolicy>
       </property>
       <property name="minimumSize">
        <size>
         <width>0</width>
         <height>50</height>
        </size>
       </property>
       <property name="title">
        <string>Исходный файл:</string>
       </property>
       <widget class="QWidget" name="layoutWidget">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>30</y>
          <width>301</width>
          <height>31</height>
         </rect>
        </property>
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <property name="sizeConstraint">
          <enum>QLayout::SetDefaultConstraint</enum>
         </property>
         <item>
          <widget class="QLineEdit" name="pathFile1">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
             <horstretch>8</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="toolTip">
            <string>Файл из которого будут браться значения для поиска</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="btnFile1">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
             <horstretch>1</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="minimumSize">
            <size>
             <width>1</width>
             <height>0</height>
            </size>
           </property>
           <property name="text">
            <string>...</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="groupBox_2">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Minimum">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="title">
        <string>Файл для сравнения:</string>
       </property>
       <widget class="QWidget" name="layoutWidget">
        <property name="geometry">
         <rect>
          <x>11</x>
          <y>30</y>
          <width>301</width>
          <height>31</height>
         </rect>
        </property>
        <layout class="QHBoxLayout" name="horizontalLayout">
         <item>
          <widget class="QLineEdit" name="pathFile2">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
             <horstretch>8</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="toolTip">
            <string>Файл в котором будет поиск</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="btnFile2">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
             <horstretch>1</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="minimumSize">
            <size>
             <width>1</width>
             <height>0</height>
            </size>
           </property>
           <property name="text">
            <string>...</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QGroupBox" name="groupBox_3">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>90</y>
      <width>321</width>
      <height>81</height>
     </rect>
    </property>
    <property name="title">
     <string>Целевая директория:</string>
    </property>
    <widget class="QWidget" name="layoutWidget">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>40</y>
       <width>301</width>
       <height>26</height>
      </rect>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout_4">
      <item>
       <widget class="QLineEdit" name="pathFile3">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
          <horstretch>8</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="toolTip">
         <string>Директория для выходного файла</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="btnOutDir">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
          <horstretch>1</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>1</width>
          <height>0</height>
         </size>
        </property>
        <property name="text">
         <string>...</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </widget>
   <widget class="QProgressBar" name="progressBar">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>363</y>
      <width>321</width>
      <height>20</height>
     </rect>
    </property>
    <property name="value">
     <number>0</number>
    </property>
    <property name="textVisible">
     <bool>false</bool>
    </property>
   </widget>
   <widget class="QLabel" name="animation">
    <property name="geometry">
     <rect>
      <x>590</x>
      <y>305</y>
      <width>51</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>180</y>
      <width>321</width>
      <height>181</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout">
     <item>
      <widget class="QGroupBox" name="groupBox_4">
       <property name="title">
        <string>Опции выборки:</string>
       </property>
       <widget class="QWidget" name="layoutWidget_2">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>60</y>
          <width>301</width>
          <height>26</height>
         </rect>
        </property>
        <layout class="QHBoxLayout" name="horizontalLayout_6">
         <item>
          <widget class="QLabel" name="label_3">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
             <horstretch>2</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>Диапазон 2:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QComboBox" name="cellAddress2">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="range_start2"/>
         </item>
         <item>
          <widget class="QLabel" name="label_4">
           <property name="text">
            <string>:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="range_end2"/>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="layoutWidget">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>20</y>
          <width>301</width>
          <height>26</height>
         </rect>
        </property>
        <layout class="QHBoxLayout" name="horizontalLayout_5">
         <item>
          <widget class="QLabel" name="label">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
             <horstretch>2</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>Диапазон 1:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QComboBox" name="cellAddress1">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="range_start1">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="toolTip">
            <string>Начальное значение напр. &quot;A1&quot;</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="label_2">
           <property name="text">
            <string>:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="range_end1">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="toolTip">
            <string>Конечное значение, напр. &quot;А22&quot;</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="layoutWidget">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>100</y>
          <width>301</width>
          <height>26</height>
         </rect>
        </property>
        <layout class="QHBoxLayout" name="horizontalLayout_8">
         <item>
          <widget class="QLabel" name="label_5">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>Лист док. 1:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QComboBox" name="comboSheet1">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
             <horstretch>1</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="layoutWidget">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>130</y>
          <width>301</width>
          <height>26</height>
         </rect>
        </property>
        <layout class="QHBoxLayout" name="horizontalLayout_9">
         <item>
          <widget class="QLabel" name="label_6">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>Лист док. 2:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QComboBox" name="comboSheet2">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
             <horstretch>1</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QGroupBox" name="groupBox_5">
    <property name="geometry">
     <rect>
      <x>660</x>
      <y>10</y>
      <width>311</width>
      <height>381</height>
     </rect>
    </property>
    <property name="title">
     <string>Терминал</string>
    </property>
    <widget class="QListWidget" name="listterminal">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>291</width>
       <height>311</height>
      </rect>
     </property>
     <property name="frameShape">
      <enum>QFrame::WinPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Sunken</enum>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>340</y>
       <width>51</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="icon">
      <iconset resource="res.qrc">
       <normaloff>:/image/delete.png</normaloff>:/image/delete.png</iconset>
     </property>
     <property name="iconSize">
      <size>
       <width>32</width>
       <height>32</height>
      </size>
     </property>
    </widget>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>340</x>
      <y>280</y>
      <width>311</width>
      <height>101</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout_7">
     <item>
      <widget class="QPushButton" name="btnStart">
       <property name="toolTip">
        <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Начать анализ файлов&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="icon">
        <iconset resource="res.qrc">
         <normaloff>:/image/analys.png</normaloff>:/image/analys.png</iconset>
       </property>
       <property name="iconSize">
        <size>
         <width>32</width>
         <height>32</height>
        </size>
       </property>
       <property name="default">
        <bool>true</bool>
       </property>
       <property name="flat">
        <bool>false</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="btnReset">
       <property name="toolTip">
        <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Начать запись выходного файла&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="icon">
        <iconset resource="res.qrc">
         <normaloff>:/image/sort.png</normaloff>:/image/sort.png</iconset>
       </property>
       <property name="iconSize">
        <size>
         <width>32</width>
         <height>32</height>
        </size>
       </property>
       <property name="default">
        <bool>true</bool>
       </property>
       <property name="flat">
        <bool>false</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="btnExit">
       <property name="toolTip">
        <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Выход из программы&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="icon">
        <iconset resource="res.qrc">
         <normaloff>:/image/exit.png</normaloff>:/image/exit.png</iconset>
       </property>
       <property name="iconSize">
        <size>
         <width>32</width>
         <height>32</height>
        </size>
       </property>
       <property name="default">
        <bool>true</bool>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QGroupBox" name="groupBox_6">
    <property name="geometry">
     <rect>
      <x>330</x>
      <y>90</y>
      <width>321</width>
      <height>161</height>
     </rect>
    </property>
    <property name="title">
     <string>Выходные параметры:</string>
    </property>
    <widget class="QCheckBox" name="checkSaveFile">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>110</y>
       <width>121</width>
       <height>22</height>
      </rect>
     </property>
     <property name="text">
      <string>Вывести в файл</string>
     </property>
    </widget>
    <widget class="QWidget" name="">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>40</y>
       <width>301</width>
       <height>26</height>
      </rect>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout_10">
      <item>
       <widget class="QLabel" name="text_filename">
        <property name="text">
         <string>Имя файла:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="filenameEdit">
        <property name="text">
         <string>outfile.xlsx</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
    <widget class="QWidget" name="">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>70</y>
       <width>301</width>
       <height>26</height>
      </rect>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout_11">
      <item>
       <widget class="QLabel" name="label_7">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
          <horstretch>2</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text">
         <string>Вывести ячейки</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QComboBox" name="comboExitCell1">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QComboBox" name="comboExitCell2">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>983</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu">
    <property name="title">
     <string>Меню</string>
    </property>
    <addaction name="action"/>
   </widget>
   <widget class="QMenu" name="menu_2">
    <property name="title">
     <string>Помощь</string>
    </property>
    <addaction name="action_2"/>
   </widget>
   <addaction name="menu"/>
   <addaction name="menu_2"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="action">
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/image/exit2.png</normaloff>:/image/exit2.png</iconset>
   </property>
   <property name="text">
    <string>Выйти</string>
   </property>
  </action>
  <action name="action_2">
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/image/about.png</normaloff>:/image/about.png</iconset>
   </property>
   <property name="text">
    <string>О программе</string>
   </property>
  </action>
 </widget>
 <tabstops>
  <tabstop>pathFile1</tabstop>
  <tabstop>pathFile2</tabstop>
  <tabstop>cellAddress1</tabstop>
  <tabstop>range_start1</tabstop>
  <tabstop>range_end1</tabstop>
  <tabstop>cellAddress2</tabstop>
  <tabstop>range_start2</tabstop>
  <tabstop>range_end2</tabstop>
  <tabstop>comboSheet1</tabstop>
  <tabstop>comboSheet2</tabstop>
  <tabstop>btnStart</tabstop>
  <tabstop>btnReset</tabstop>
  <tabstop>btnExit</tabstop>
  <tabstop>pushButton</tabstop>
  <tabstop>listterminal</tabstop>
  <tabstop>btnFile2</tabstop>
  <tabstop>btnFile1</tabstop>
  <tabstop>pathFile3</tabstop>
  <tabstop>btnOutDir</tabstop>
 </tabstops>
 <resources>
  <include location="res.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>btnFile1</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>s_getPath1()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>312</x>
     <y>80</y>
    </hint>
    <hint type="destinationlabel">
     <x>485</x>
     <y>120</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btnFile2</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>s_getPath2()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>628</x>
     <y>83</y>
    </hint>
    <hint type="destinationlabel">
     <x>630</x>
     <y>116</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btnOutDir</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>s_getPath3()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>311</x>
     <y>166</y>
    </hint>
    <hint type="destinationlabel">
     <x>418</x>
     <y>211</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btnStart</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>s_start()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>439</x>
     <y>312</y>
    </hint>
    <hint type="destinationlabel">
     <x>473</x>
     <y>161</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btnExit</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>649</x>
     <y>317</y>
    </hint>
    <hint type="destinationlabel">
     <x>632</x>
     <y>331</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btnExit</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>s_exit()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>649</x>
     <y>312</y>
    </hint>
    <hint type="destinationlabel">
     <x>608</x>
     <y>244</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btnReset</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>s_begin()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>479</x>
     <y>332</y>
    </hint>
    <hint type="destinationlabel">
     <x>483</x>
     <y>295</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton</sender>
   <signal>clicked()</signal>
   <receiver>listterminal</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>698</x>
     <y>387</y>
    </hint>
    <hint type="destinationlabel">
     <x>740</x>
     <y>287</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>s_getPath1()</slot>
  <slot>s_getPath2()</slot>
  <slot>s_getPath3()</slot>
  <slot>s_start()</slot>
  <slot>s_exit()</slot>
  <slot>about()</slot>
  <slot>s_begin()</slot>
 </slots>
</ui>
